### ISSUE TITLE
Duplicite and empty translations field

### ISSUE THREAD

**Issue by safrataxyz** [directus/directus#14331]
### Describe the Bug

Hello,

I am reporting an issue I noticed a couple of times, which is present in translations field.
In our project we use only two languages, Czech and English, but a third field appeared, with null languages_code and text from "en" field. See the result, that comes back from the API:

```
{
	"id": 3,
	"internal_name": "VLOS",
	"translations": [
		{
			"id": 5,
			"questionnaire_result_rules_id": 3,
			"languages_code": "cs",
			"title": "Vizuální dohled",
			"icon": "vlos",
			"text": "<div>\n<div>Během letu udržujte dron ve vizu&aacute;ln&iacute;m dohledu.</div>\n</div>"
		},
		{
			"id": 6,
			"questionnaire_result_rules_id": 3,
			"languages_code": null,
			"title": null,
			"icon": null,
			"text": "<div>\n<div>Fly only within the visual line of sight.</div>\n</div>"
		},
		{
			"id": 7,
			"questionnaire_result_rules_id": 3,
			"languages_code": "en",
			"title": "Watch Your Drone",
			"icon": "vlos",
			"text": "<div>Fly only within the visual line of sight.</div>"
		}
	]
},
```

In the table I see also all three translations:
![image](https://user-images.githubusercontent.com/42304713/177991228-a7906610-79e9-4666-bd17-19e18682c5e2.png)

But when I click inside, only Czech and English fields, are correctly shown:
![image](https://user-images.githubusercontent.com/42304713/177991422-27e358dd-ae5b-4a62-90dd-065146c6a57f.png)

### To Reproduce

Not sure what has caused this, so I cannot really say how to reproduce, I was making this item same as every other item, but this one had this issue.

### Errors Shown

None relevant

### What version of Directus are you using?

9.13.0

### What version of Node.js are you using?

16.15.1 - Official docker image from Dockerhub

### What database are you using?

Postgres 13.4

### What browser are you using?

Brave

### How are you deploying Directus?

Docker

**Comment by br41nslug** [comment]
This was fixed in https://github.com/directus/directus/pull/14300 and released with directus v9.14.0 yesterday 😄 
We can re-open the ticket if the problem persists after updating

**Comment by safrataxyz** [comment]
Just updated to v9.14.1 and the problem still persists

**Comment by marianhlavac** [comment]
I guess that the fix in #14300 won't actually remove the `null` entries in the database, but should prevent them to appear ever again, am I right?

**Comment by rijkvanzanten** [comment]
> I guess that the fix in #14300 won't actually remove the `null` entries in the database, but should prevent them to appear ever again, am I right?

That's correct
